File: pdlearning/src/microservices/learner-microservice/Api/Application/Models/TrackingModel.cs
using System;
using Microservice.Learner.Domain.ValueObject;

namespace Microservice.Learner.Application.Models
{
    public class TrackingModel
    {
        public Guid ItemId { get; set; }

        public LearningTrackingType ItemType { get; set; }

        public bool IsLike { get; set; }

        public int TotalLike { get; set; }

        public int TotalShare { get; set; }

        public int TotalView { get; set; }

        public int TotalDownload { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/learner/models/tracking.model.ts
export interface ITrackingModel {
  itemId: string;
  itemType: string;
  isLike: boolean;
  totalLike: number;
  totalShare: number;
  totalView: number;
  totalDownload: number;
}

export class TrackingModel implements ITrackingModel {
  public itemId: string;
  public itemType: string;
  public isLike: boolean;
  public totalLike: number;
  public totalShare: number;
  public totalView: number;
  public totalDownload: number;
  constructor(data?: ITrackingModel) {
    if (data) {
      this.itemId = data.itemId;
    }
    this.itemType = data.itemType;
    this.isLike = data.isLike;
    this.totalLike = data.totalLike;
    this.totalShare = data.totalShare;
    this.totalView = data.totalView;
    this.totalDownload = data.totalDownload;
  }
}
